core/commonMain/src/kotlinx/serialization/internal/BuiltInSerializers.kt

```diff
@@ -37,3 +38,17 @@ internal object NothingSerializer : KSerializer<Nothing> {
         throw SerializationException("'kotlin.Nothing' does not have instances")
     }
 }
+
+@PublishedApi
+@ExperimentalStdlibApi
+internal object UuidSerializer: KSerializer<Uuid> {
+    override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("kotlin.uuid.Uuid", PrimitiveKind.STRING)
+
+    override fun serialize(encoder: Encoder, value: Uuid) {
+        encoder.encodeString(value.toString())
+    }
+
+    override fun deserialize(decoder: Decoder): Uuid {
+        return Uuid.parse(decoder.decodeString())
```

Comment:
Shouldn't we wrap exceptions is `serialize` and `deserialize` to throw only SerializationException?

Reply:
We are allowed to throw IAE on errors not related to decoding: https://github.com/Kotlin/kotlinx.serialization/blob/be99c0d780712b47144be528222377575e9b21ff/core/commonMain/src/kotlinx/serialization/SerialFormat.kt#L101